src/linear_algebra/orientation.lean

```diff
@@ -0,0 +1,384 @@
+/-
+Copyright (c) 2021 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import linear_algebra.determinant
+
+/-!
+# Orientations of modules and rays in modules
+
+This file defines rays in modules and orientations of modules.
+
+## Main definitions
+
+* `module.ray` is a type for the equivalence class of nonzero vectors in a module with some
+common positive multiple.
+
+* `orientation` is a type synonym for `module.ray` for the case where the module is that of
+alternating maps from a module to its underlying ring.  An orientation may be associated with an
+alternating map or with a basis.
+
+* `module.oriented` is a type class for a choice of orientation of a module that is considered
+the positive orientation.
+
+## Implementation notes
+
+`orientation` is defined for an arbitrary index type, but the main intended use case is when
```

Comment:
I think this remark should also be part of the doc string for `orientation`.

Reply:
That doc string already says "intended to be used when `ι` is a `fintype` with the same cardinality as a basis.", is that not sufficient?

Reply:
It is sufficient, I must be going blind. Sorry for the noise!